File: myExpenses/src/test/java/org/totschnig/myexpenses/testutils/MockLicenceModule.java
package org.totschnig.myexpenses.testutils;

import com.google.android.vending.licensing.Obfuscator;

import org.totschnig.myexpenses.MyApplication;
import org.totschnig.myexpenses.di.LicenceModule;

public class MockLicenceModule extends LicenceModule {

  @Override
  protected String provideDeviceId(MyApplication application) {
    return "DUMMY";
  }

  @Override
  protected Obfuscator provideObfuscator(String deviceId, MyApplication application) {
    return new Obfuscator() {
      @Override
      public String obfuscate(String original, String key) {
        return original;
      }

      @Override
      public String unobfuscate(String obfuscated, String key) {
        return obfuscated;
      }
    };
  }
}

File: myExpenses/src/debug/java/org/totschnig/myexpenses/testutils/MockLicenceModule.kt
package org.totschnig.myexpenses.testutils

import com.google.android.vending.licensing.Obfuscator
import com.google.android.vending.licensing.PreferenceObfuscator
import org.totschnig.myexpenses.MyApplication
import org.totschnig.myexpenses.db2.Repository
import org.totschnig.myexpenses.di.LicenceModule
import org.totschnig.myexpenses.preference.PrefHandler
import org.totschnig.myexpenses.util.ICurrencyFormatter
import org.totschnig.myexpenses.util.crashreporting.CrashHandler
import org.totschnig.myexpenses.util.licence.LicenceHandler
import java.time.Clock

class MockLicenceModule(private val clock: Clock) : LicenceModule() {
    override fun provideDeviceId(application: MyApplication) = "DUMMY"

    override fun provideObfuscator(deviceId: String, application: MyApplication) =
        object : Obfuscator {
            override fun obfuscate(original: String, key: String): String {
                return original
            }

            override fun unobfuscate(obfuscated: String, key: String): String {
                return obfuscated
            }
        }

    override fun providesLicenceHandler(
        preferenceObfuscator: PreferenceObfuscator,
        crashHandler: CrashHandler,
        application: MyApplication,
        prefHandler: PrefHandler,
        repository: Repository,
        currencyFormatter: ICurrencyFormatter
    ): LicenceHandler = MockLicenceHandler(application, preferenceObfuscator, crashHandler, prefHandler, repository, currencyFormatter, clock)
}